### ISSUE TITLE
SU header issue

### ISSUE THREAD

**Issue by mo7sen87** [geodynamics/specfem2d#1037]
I am a new user to speckfm2d. I am modeling a very high frequency full wavefield sonic experiment. To fulfill the stability measures, I am using dt = 0.0000001 a grid of 1 cm in X and Z direction. Whenever I record nt >50000 I have an issue with the SU headers. Whenever I try to plot the SU data I get the following error (fetter.c:on trace#2 number of samples in header differs from number for first trace). I don't see this error when I use number of samples <50000. What may be the cause of the problem

**Comment by ICNSK** [comment]
I find `SU_FORMAT = .true.` option producing unusable files, `save_binary_seismograms_single  = .true.` work better than ascii for dense set of stations, it produce binary file with number of samples equal to `NSTEP`. If could be converted to su using 

`suaddhead n1=$NSTEP  < Uz_file_single_d.bin | sushw key=dt,delrt a=$DT,$start_time  > Uz_file_single_d.su`

where `$NSTEP`  and `$DT` are from `Par_file`, and `$start_time` is (unfortunately) from log file section "time loop"
`$DT` and `$start_time` need to be in  microseconds = 1e-6 sec type `unsigned_short `  and number of samples is limited to `unsigned_short_max = 65535`  according to [source code](https://github.com/JohnWStockwellJr/SeisUnix/blob/master/include/segy.h)  but  su does not allows float headers, so with your timestep  0.0000001 = 1e-7 = 1e-1 microsec you will not be able  to create valid su file, however you could still visualize simulations using ximage

`ximage n1=$NSTEP d1=$DT f1=$start_time < Uz_file_single_d.bin perc=99`

**Comment by danielpeter** [comment]
yes, the SU format is probably not the right format in this case. 

note that the implemented SU header uses short integer, thus the number of timesteps and/or number of receivers is limited to 32767.

**Comment by kawaalina** [comment]
root@DESKTOP-6TE0R0N:~# ximage
Command 'ximage' not found

why i have this??

**Comment by ICNSK** [comment]
ximage is a part of the Seismic Unix package (https://github.com/JohnWStockwellJr/SeisUnix). You need to download, compile and install it following instructions from https://github.com/JohnWStockwellJr/SeisUnix/blob/master/src/README_TO_INSTALL 
You might need to edit Makefile.config, depending on your system and install some required packages. 
Checkout https://hadi-tim.github.io/blog/2022/seismic-unix/

**Comment by kawaalina** [comment]
it's work, thanx :>
but if i want to add new equation to par_file how can i do that? whare is the function base folders?
